patches/linux/arch-4-s390x.patch

```diff
@@ -0,0 +1,465 @@
+ 					Platform::LinuxLoong64
++				} else if cfg!(target_arch = "s390x") {
++					Platform::LinuxS390X
+ 				} else {
+diff --git a/resources/server/bin/helpers/check-requirements-linux.sh b/resources/server/bin/helpers/check-requirements-linux.sh
+index 8ef07a2..00dd3e2 100644
+--- a/resources/server/bin/helpers/check-requirements-linux.sh
++++ b/resources/server/bin/helpers/check-requirements-linux.sh
+@@ -59,2 +59,3 @@ case $ARCH in
+     loongarch64) LDCONFIG_ARCH="double-float";;
++    s390x)   LDCONFIG_ARCH="64bit";;
+ esac
+diff --git a/src/vs/platform/extensionManagement/common/extensionManagement.ts b/src/vs/platform/extensionManagement/common/extensionManagement.ts
+index ea7b30d..004df84 100644
+--- a/src/vs/platform/extensionManagement/common/extensionManagement.ts
++++ b/src/vs/platform/extensionManagement/common/extensionManagement.ts
+@@ -44,2 +44,3 @@ export function TargetPlatformToString(targetPlatform: TargetPlatform) {
+ 		case TargetPlatform.LINUX_LOONG64: return 'Linux Loong64';
++		case TargetPlatform.LINUX_S390X: return 'Linux S390X';
+ 
+@@ -70,2 +71,3 @@ export function toTargetPlatform(targetPlatform: string): TargetPlatform {
+ 		case TargetPlatform.LINUX_LOONG64: return TargetPlatform.LINUX_LOONG64;
++		case TargetPlatform.LINUX_S390X: return TargetPlatform.LINUX_S390X;
+ 
+@@ -114,2 +116,5 @@ export function getTargetPlatform(platform: Platform | 'alpine', arch: string |
+ 			}
++			if (arch === 's390x') {
++				return TargetPlatform.LINUX_S390X;
++			}
+ 			return TargetPlatform.UNKNOWN;
+diff --git a/src/vs/platform/extensions/common/extensions.ts b/src/vs/platform/extensions/common/extensions.ts
+index 5973d4d..a7d6298 100644
+--- a/src/vs/platform/extensions/common/extensions.ts
++++ b/src/vs/platform/extensions/common/extensions.ts
+@@ -297,2 +297,3 @@ export const enum TargetPlatform {
+ 	LINUX_LOONG64 = 'linux-loong64',
++	LINUX_S390X = 'linux-s390x',
+ 
+diff --git a/package-lock.json b/package-lock.json
+index 7b02996..4642486 100644
+--- a/package-lock.json
++++ b/package-lock.json
+@@ -17875,10 +17875,11 @@
+       }
+     },
+     "node_modules/web-tree-sitter": {
+-      "version": "0.20.8",
+-      "resolved": "https://registry.npmjs.org/web-tree-sitter/-/web-tree-sitter-0.20.8.tgz",
+-      "integrity": "sha512-weOVgZ3aAARgdnb220GqYuh7+rZU0Ka9k9yfKtGAzEYMa6GgiCzW9JjQRJyCJakvibQW+dfjJdihjInKuuCAUQ==",
+-      "dev": true
++      "version": "0.23.0",
++      "resolved": "https://registry.npmjs.org/web-tree-sitter/-/web-tree-sitter-0.23.0.tgz",
++      "integrity": "sha512-p1T+ju2H30fpVX2q5yr+Wv/NfdMMWMjQp9Q+4eEPrHAJpPFh9DPfI2Yr9L1f5SA5KPE+g1cNUqPbpihxUDzmVw==",
++      "dev": true,
++      "license": "MIT"
+     },
+     "node_modules/webidl-conversions": {
+       "version": "3.0.1",
+diff --git a/package.json b/package.json
+index 7e0de18..4ceaded 100644
+--- a/package.json
++++ b/package.json
+@@ -226,7 +226,10 @@
+     },
+     "@parcel/watcher@2.1.0": {
+       "node-addon-api": "7.1.0"
++    },
```

Comment:
You can't touch the `package.json` since it will affect the others platforms and even Windows.

Reply:
unfortunately that change is required because the version pulled in by @vscode/l10n-dev causes the build to fail because of an endianness issue. However, that dependency change to package.json should not be a breaking change for any other architecture or platform.

Do you have a suggestion how that change could be limited to the s390x build only?

Reply:
I've just patched the `package_linux_reh.sh` to include patches in `patches/linux/reh/${VSCODE_ARCH}/`.

The build in the CI is a bit cumbersome:
- the first step is to compile the project from TS to JS on x86
- the JS files are then dispatched to each architectures
- the build for an arch is installed the dependencies for the arch

From the current version, it's also how it's done on Windows (built on linux x86) then packaged on Windows. MS does it the same way since at least a year.

By adding a patch in `patches/linux/reh/${VSCODE_ARCH}/`, I'm hoping that the dependencies are correctly installed for your arch.

Reply:
thank you, I have rebased my changes and moved the part of the patch that changes package.json to that subdirectory. CI build is running now, I'll comment here again when that finishes.